code/modules/mob/living/carbon/human/species/slime.dm

```diff
@@ -85,6 +85,11 @@
 
 /datum/species/slime/handle_life(mob/living/carbon/human/H)
 	// Slowly shifting to the color of the reagents
+	if((H.blood_color != BlendRGB(H.skin_colour, "#acacac", 0.5))||H.blood_color == null) // Put here, so if it's a roundstart, dyed, or CMA'd slime, their blood changes to match skin
+		if(H.blood_color == null)
+			H.blood_color = H.skin_colour //Just to get it not null so the math works
+		H.blood_color = BlendRGB(H.skin_colour, "#acacac", 0.5) // Blends this to make it work better
+		H.dna.species.blood_color = H.blood_color
```

Comment:
I'd have these all included into the check below and have the `blood_color` get properly initialized once the mob color is known. Doing this will save on processing done each life tick since it's only done when the skin color changes

Reply:
Used signals as requested over discord